Rakefile

```diff
@@ -71,6 +75,63 @@ def re_run_basic_smoke_test
+  JSON.parse(@memoized_passing_json)['actions'][0]['parameters'].find{|x| x['name'] == 'SUITE_COMMIT'}['value']
+end
+def get_passing_puppet_sha(current_branch_trunk)
+  get_passing_json(current_branch_trunk)
+  JSON.parse(@memoized_passing_json)['actions'][0]['parameters'].find{|x| x['name'] == 'puppet_COMPONENT_COMMIT'}['value']
+end
+
+PUPPET_SUBMODULE_PATH = File.join('ruby','puppet')
+def get_passing_puppet_version(current_branch_trunk)
+   #FIXME: this should be updated when the package yaml file contains the metadata we need
+   #  we have to replace the hyphens with dots because, vanagon
+  `cd #{PUPPET_SUBMODULE_PATH}; git describe`.strip.gsub(/-/,'.')
+end
+
+def replace_puppet_pins(passing_puppetagent_sha, passing_puppet_version)
+  # read beaker options hash from its file
+  beaker_options_file_path = 'acceptance/config/beaker/options.rb'
+  puts("replacing puppet sha and version in #{beaker_options_file_path}, with agent sha: #{get_passing_puppetagent_sha(current_branch_trunk)} and puppet version: #{get_passing_puppet_version(current_branch_trunk).to_s}")
+  beaker_options_from_file = eval(File.read(beaker_options_file_path))
+  # add puppet version values
+  beaker_options_from_file[:puppet_version]       = passing_puppet_version
+  beaker_options_from_file[:puppet_build_version] = passing_puppetagent_sha
+  File.write(beaker_options_file_path, beaker_options_from_file.pretty_inspect)
+end
+
+namespace :puppet_submodule do
+  puppet_pins_file_path = File.join('acceptance','lib','helper.rb')
+  desc 'update puppet submodule commit'
+  task :update_puppet_version do
+    #hard code current branch
+    #  it's too difficult to reliably find trunk of a feature branch
+    #  ugh... now jenkins job naming is out of date with new component branches
+    current_branch_trunk = '5.0.x'
+    #update puppet submodule
+    #  ensure we fetch here, or the describe done later could be wrong
+    git_checkout_command = "cd #{PUPPET_SUBMODULE_PATH} && git fetch origin && git checkout #{get_passing_puppet_sha(current_branch_trunk)}"
+    puts("checking out known passing puppet version in submodule: `#{git_checkout_command}`")
+    system(git_checkout_command)
+    # replace puppet version and sha pins in:
+    replace_puppet_pins(get_passing_puppetagent_sha(current_branch_trunk), get_passing_puppet_version(current_branch_trunk))
+  end
+  desc 'commit and push; CAUTION: WILL commit and push, upstream, local changes to the puppet submodule and acceptance helper'
+  task :commit_push do
+    #  commit and push
+    puts("commiting and pushing submodule and pins via: `git add #{PUPPET_SUBMODULE_PATH}; git add #{puppet_pins_file_path}; git commit -m '(maint) update puppet submoudule version and pins'`")
+    system("git checkout #{current_branch_trunk} && git add #{PUPPET_SUBMODULE_PATH} && git add #{puppet_pins_file_path} && git commit -m '(maint) update puppet submoudule version and pins' && git push origin")
```

Comment:
If we're using `current_branch_trunk` in multiple places I would probably then turn it into a constant and move it to the top of the file.